docs/table-dictionary-length.md

```diff
@@ -0,0 +1,33 @@
+# Table Dictionary Length
+
+## Summary
+
+A built-in function to get the length of the dictionary part of a table.
+
+## Motivation
+
+An O(1) method to count the number of elements in the dictionary part of the table by lifting the information directly from the underlying data structure.  Without this, it's only possible to get the count by premeditated management of a count variable or by outright counting on the spot.
```

Comment:
I would like to see a motivation that isn't abstract. Right now, I could just tell you to write something with the type `{{k: string, v: unknown}}` and it would be O(1) and works with `nil`.